src/Controls/src/Core/Platform/Android/Extensions/ButtonExtensions.cs

```diff
@@ -41,19 +41,30 @@ public static void UpdateContentLayout(this MaterialButton materialButton, Butto
 				// ourselves for our custom implemented IconGravityBottom
 				materialButton.IconPadding = (int)context.ToPixels(contentLayout.Spacing);
 
+				// For IconGravityTextEnd and IconGravityTextStart, setting the Icon twice
+				// is needed to work around the Android behavior that caused
+				// https://github.com/dotnet/maui/issues/11755
 				switch (contentLayout.Position)
 				{
 					case ButtonContentLayout.ImagePosition.Top:
+						materialButton.Icon = null;
 						materialButton.IconGravity = MaterialButton.IconGravityTop;
+						materialButton.Icon = icon;
 						break;
 					case ButtonContentLayout.ImagePosition.Bottom:
+						materialButton.Icon = null;
+						TextViewCompat.SetCompoundDrawablesRelative(materialButton, null, null, null, icon);
```

Comment:
Why is this needed? I do this in the OnLayout:

https://github.com/dotnet/maui/blob/05078ba9836c48266fefd674c18c566248c2b7fa/src/Core/src/Platform/Android/MauiMaterialButton.cs#L55-L63